torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6962,6 +6962,18 @@ def make_input():
 
     return samples
 
+def sample_inputs_tensorsolve(op_info, device, dtype, requires_grad, **kwargs):
+    a_shapes = [(2, 3, 6), (3, 4, 4, 3)]
+    dimss = [None, (0, 2)]
+
+    def gen_inputs():
+        for a_shape, dims in itertools.product(a_shapes, dimss):
```

Comment:
This is fine but just a few handwritten inputs that test interesting cases would be a great alternative

Reply:
I wanted to add also shapes with 0, but NumPy fails then. It's difficult to be creative here: this function reshapes the input to 2D and calls `torch.linalg.solve` internally; besides 0 in the shape nothing else seems interesting.

Reply:
Adding zero would be interesting. I'd at least add the case and comment it out with a note about the NumPy failure. Do you want to report a bug to NumPy?

Reply:
I added a zero dim case and submitted a bug fix to NumPy. With the bug fix, all tests pass for sample input with zero dimensions.